From 24ca33edd4986982fe0a30409f850bc089013681 Mon Sep 17 00:00:00 2001
From: Pete Moore <pmoore@mozilla.com>
Date: Sun, 20 Apr 2025 12:17:56 +0200
Subject: [PATCH] log mmio reads/writes

---
 include/asm-generic/io.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/include/asm-generic/io.h b/include/asm-generic/io.h
index 80de699bf6af..75835a6bc837 100644
--- a/include/asm-generic/io.h
+++ b/include/asm-generic/io.h
@@ -94,11 +94,15 @@ void log_post_read_mmio(u64 val, u8 width, const volatile void __iomem *addr,
 static inline void log_write_mmio(u64 val, u8 width, volatile void __iomem *addr,
 				  unsigned long caller_addr, unsigned long caller_addr0) {}
 static inline void log_post_write_mmio(u64 val, u8 width, volatile void __iomem *addr,
-				       unsigned long caller_addr, unsigned long caller_addr0) {}
+				       unsigned long caller_addr, unsigned long caller_addr0) {
+	printk(KERN_INFO "Write %u bits [0x%px]=%#018llx\n", width, addr, val);
+}
 static inline void log_read_mmio(u8 width, const volatile void __iomem *addr,
 				 unsigned long caller_addr, unsigned long caller_addr0) {}
 static inline void log_post_read_mmio(u64 val, u8 width, const volatile void __iomem *addr,
-				      unsigned long caller_addr, unsigned long caller_addr0) {}
+				      unsigned long caller_addr, unsigned long caller_addr0) {
+	printk(KERN_INFO "Read %u bits [0x%px]=%#018llx\n", width, addr, val);
+}
 
 #endif /* CONFIG_TRACE_MMIO_ACCESS */
 
-- 
2.47.0

